mgmt-lambda/app.ts

```diff
@@ -1,175 +1,66 @@
+import { defaults } from './DefaultSettings'
+import { handleStatus } from './handlers/statusHandler'
+import { handleUpdate } from './handlers/updateHandler'
+import { LambdaClient } from '@aws-sdk/client-lambda'
+import { CloudFrontClient } from '@aws-sdk/client-cloudfront'
+
+export async function handler(event: APIGatewayProxyEventV2WithRequestContext<APIGatewayEventRequestContextV2>) {
+  const secretManagerClient = new SecretsManagerClient({ region: defaults.AWS_REGION })
 
-const REGION = 'us-east-1'
-
-export async function handler(event: any, ctx: any) {
-  console.info(JSON.stringify(event))
-
-  const job = event['CodePipeline.job']
-  if (!job) {
-    console.error('No job found')
-    return
-  }
-
-  const userInput = JSON.parse(job.data.actionConfiguration.configuration.UserParameters)
-  const lambdaFunctionName = userInput.LAMBDA_NAME
-  const cloudFrontDistrId = userInput.CF_DISTR_ID
-
-  console.info(`Going to upgrade Fingerprint Pro function association at CloudFront distbution.`)
-  console.info(`Lambda function: ${lambdaFunctionName}. CloudFront ID: ${cloudFrontDistrId}`)
-
-  const latestFunctionArn = await getLambdaLatestVersionArn(lambdaFunctionName)
-  if (!latestFunctionArn) {
-    return publishJobFailure(ctx, job, 'No lambda versions')
-  }
-
-  if (latestFunctionArn.length === 1) {
-    console.info('No updates yet')
-    return publishJobSuccess(ctx, job)
-  }
-
-  const cloudFrontClient = new CloudFrontClient({ region: REGION })
-
-  const configParams = {
-    Id: cloudFrontDistrId,
-  }
-  const getConfigCommand = new GetDistributionConfigCommand(configParams)
-  const cfConfig: GetDistributionConfigCommandOutput = await cloudFrontClient.send(getConfigCommand)
-
-  if (!cfConfig.ETag || !cfConfig.DistributionConfig) {
-    return publishJobFailure(ctx, job, 'CloudFront distribution not found')
-  }
-
-  const cacheBehaviors = cfConfig.DistributionConfig.CacheBehaviors
-  const fpCbs = cacheBehaviors?.Items?.filter((it) => it.TargetOriginId === 'fpcdn.io')
-  if (!fpCbs || fpCbs?.length === 0) {
-    return publishJobFailure(ctx, job, 'Cache behavior not found')
-  }
-  const cacheBehavior = fpCbs[0]
-  const lambdas = cacheBehavior.LambdaFunctionAssociations?.Items?.filter(
-    (it) => it && it.EventType === 'origin-request' && it.LambdaFunctionARN?.includes(lambdaFunctionName),
-  )
-  if (!lambdas || lambdas?.length === 0) {
-    return publishJobFailure(ctx, job, 'Lambda function association not found')
-  }
-  const lambda = lambdas[0]
-  lambda.LambdaFunctionARN = latestFunctionArn
-
-  const updateParams: UpdateDistributionCommandInput = {
-    DistributionConfig: cfConfig.DistributionConfig,
-    Id: cloudFrontDistrId,
-    IfMatch: cfConfig.ETag,
+  try {
+    const authSettings = await getAuthSettings(secretManagerClient)
+    const authorization = event.headers['authorization']
+    if (authorization !== authSettings.token) {
```

Comment:
Will `authorization` header be directly the token, no scheme? This header needs [a scheme](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization#syntax) by standard, such as `Basic` or something, no?

Reply:
So what will we do about this in this PR?